plugins/backup/networker/src/main/java/org/apache/cloudstack/backup/NetworkerBackupProvider.java

```diff
@@ -0,0 +1,640 @@
+        BackupOfferingVO vmBackupOffering = new BackupOfferingDaoImpl().findById(vm.getBackupOfferingId());
+        final String backupProviderPolicyId = vmBackupOffering.getExternalId();
+        String backupRentionPeriod = getClient(vm.getDataCenterId()).getBackupPolicyRetentionInterval(backupProviderPolicyId);
+
+        if ( backupRentionPeriod == null ) {
+            LOG.warn("There is no retention setting for Emc Networker Policy, setting default for 1 day");
+            backupRentionPeriod="1 Day";
+        }
+
+        // Get Cluster
+        clusterName = getVMHypervisorCluster(hostVO);
+
+        String command = "sudo /usr/share/cloudstack-common/scripts/vm/hypervisor/kvm/nsrkvmbackup.sh" +
+                " -s " + networkerServer +
+                " -R '" + backupRentionPeriod + "'" +
+                " -P " + NetworkerMediaPool.valueIn(vm.getDataCenterId()) +
+                " -c " + clusterName +
+                " -u " + vm.getUuid() +
+                " -t " + vm.getName();
+        if (NetworkerClientVerboseLogs.value())
+            command = command + " -v ";
+
+        LOG.debug("Starting backup for VM ID " + vm.getUuid() + " on Networker provider");
+        Date backupJobStart = new Date();
+
+        String saveTime = executeBackupCommand(hostVO, credentials.first(), credentials.second(), credentials.third(), command);
+        BackupVO backup = getClient(vm.getDataCenterId()).registerBackupForVm(vm, backupJobStart, saveTime);
+        if (backup != null) {
+            backupDao.persist(backup);
+            return true;
+        } else {
+            // We need to handle this rare situation where backup is successful but can't be registered properly.
+            return false;
+        }
+    }
+
+    @Override
+    public boolean deleteBackup(Backup backup) {
+
+        final Long zoneId = backup.getZoneId();
+        final String externalBackupId = backup.getExternalId();
+
+        if (getClient(zoneId).deleteBackupForVM(externalBackupId)) {
+            LOG.debug("EMC Networker successfully deleted backup with id " + externalBackupId);
+            return true;
+        } else {
+            LOG.debug("There was an error removing the backup with id " + externalBackupId + " from EMC NEtworker");
+        }
+        return false;
+    }
+
+    @Override
+    public Map<VirtualMachine, Backup.Metric> getBackupMetrics(Long zoneId, List<VirtualMachine> vms) {
+        final Map<VirtualMachine, Backup.Metric> metrics = new HashMap<>();
+        Long vmBackupSize=0L;
+        Long vmBackupProtectedSize=0L;
+
+        if (CollectionUtils.isEmpty(vms)) {
+            LOG.warn("Unable to get VM Backup Metrics because the list of VMs is empty.");
+            return metrics;
+        }
+
+        for (final VirtualMachine vm : vms) {
+            for ( Backup.VolumeInfo thisVMVol : vm.getBackupVolumeList()) {
+                    vmBackupSize += (thisVMVol.getSize() / 1024L / 1024L);
```

Comment:
should this be divided by 1024.0 then convert to Long?

Reply:
1024L is supposed to do that. vmBackupSize is already Long.